src/ngMeta.js

```diff
@@ -276,8 +283,37 @@
         return this;
       };
 
+      /**
+       * @ngdoc method
+       * @name ngMetaProvider#mergeNestedStateData
+       * @param {string} mergeNestedStateData [Optional] method to deep merge
+       * meta data for nested views.
+       *
+       * @description
+       * Helper function to Extend $stateProvider.decorator('state') to merge nested
+       * view meta data if using ui-router.
+       *
+       * @returns {Object} data
+       */
+      this.mergeNestedStateData = function(state, parentDecoratorFn) {
+      // original data
+        var originalData = parentDecoratorFn(state) || {};
+
+        // create new merged meta
+        var parentMetaData = state.parent && state.parent.data && state.parent.data.meta;
+        var mergedMeta = angular.merge({}, parentMetaData, originalData.meta);
```

Comment:
Shouldn't this statement be in the if block below? Calling angular.merge when both `parentMetaData` and `originalData.meta` are null/undefined wouldn't serve a purpose anyway.